File: 187. Repeated DNA Sequences.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):
        """
        :type s: str
        :rtype: List[str]
        """
        dic = {}
        res = []
        for i in xrange(len(s) - 9):
            if s[i:i+10] in dic:
                dic[s[i:i+10]] += 1
                if dic[s[i:i+10]] == 2:
                    res += [s[i:i+10]]
            else:
                dic[s[i:i+10]] = 1
        return res

File: 187. Repeated DNA Sequences.cpp
#include <unordered_map>
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        if (s.length() < 10) return {};
        unordered_map <string, int> dic;
        vector <string> res;
        for (int i = 0; i < s.length() - 9; i++)
        {
            string temp = s.substr(i, 10);
            if (dic.count(temp) > 0)
            {
                dic[temp]++;
                if (dic[temp] == 2) res.push_back(temp);
            }
            else
                dic[temp] = 1;
        }
        return res;
    }
};
